Rubberduck.Refactorings/EncapsulateField/FieldCandidates/ArrayCandidate.cs

```diff
@@ -1,64 +1,15 @@
-﻿using Rubberduck.Parsing;
-using Rubberduck.Parsing.Grammar;
+﻿using Rubberduck.Parsing.Grammar;
 using Rubberduck.Parsing.Symbols;
-using Rubberduck.Resources;
-using System.Linq;
 
 namespace Rubberduck.Refactorings.EncapsulateField
 {
-    public interface IArrayCandidate : IEncapsulateFieldCandidate
+    public class ArrayCandidate : EncapsulateFieldCandidate
     {
-        string UDTMemberDeclaration { get;}
-        bool HasExternalRedimOperation(out string errorMessage);
-    }
-
-    public class ArrayCandidate : EncapsulateFieldCandidate, IArrayCandidate
-    {
-        private string _subscripts;
-        public ArrayCandidate(Declaration declaration, IValidateVBAIdentifiers validator)
-            :base(declaration, validator)
+        public ArrayCandidate(Declaration declaration)
+            :base(declaration)
         {
-            ImplementLet = false;
-            ImplementSet = false;
             PropertyAsTypeName = Tokens.Variant;
```

Comment:
Arrays can be typed to something besides `Variant`? Why are we setting it `Variant` here and not looking at declaration's type?

Reply:
AIUI, we will lose Office97 back compatibility :smirk:.  Yes, it would be better to return the array's declared type.  This falls a bit outside the goal of the PR, but looks fairly straightforward to change.

Reply:
Hmm, can’t tell if we still can support O97 (we require .NET 4.6.1, and that doesn’t work on older OSes, think O97 still can be installed on W10, but unsure). Assuming yes, I am fine with leaving it as is with a comment for a future PR to conditionally branch based on Office version.

Reply:
Comment added in latest sync